src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -207,6 +225,40 @@ static async ValueTask<QuicConnection> StartConnectAsync(QuicClientConnectionOpt
     /// </summary>
     public IPEndPoint LocalEndPoint => _localEndPoint;
 
+    private async void OnStreamCapacityIncreased(int bidirectionalIncrement, int unidirectionalIncrement)
+    {
+        // Bail out early to avoid queueing work on the thread pool as well as event args instantiation.
+        if (_streamCapacityCallback is null)
+        {
+            return;
+        }
```

Comment:
Will a call to this function allocate a state machine even when no callbacks are registered? Maybe we can delegate the latter half of the function to nested private helper function.

Reply:
It will "allocate" the state machine struct on the stack until the first `await` that yields.
In this case if `_streamCapacityCallback` is null, it won't allocate anything on the heap.

Reply:
So I get that it's not worth any change :wink: